pkg/cmd/admin/router/router.go

```diff
@@ -642,12 +654,12 @@ func RunCmdRouter(f *clientcmd.Factory, cmd *cobra.Command, out io.Writer, cfg *
 		if cfg.SecretsAsEnv {
 			env.Add(app.Environment{"DEFAULT_CERTIFICATE": string(defaultCert)})
 		} else {
-			// TODO: make --credentials create secrets and bypass service account
 			env.Add(app.Environment{"DEFAULT_CERTIFICATE_PATH": defaultCertificatePath})
 		}
 	}
-
-	secrets, volumes, mounts, err := generateSecretsConfig(cfg, kClient, namespace, defaultCert)
+	env.Add(app.Environment{"DEFAULT_CERTIFICATE_DIR": defaultCertificateDir})
```

Comment:
Because we are getting a secret created for us by the annotation on the service.  BUT the secret comes in as a cert and key... not as a combined PEM.  And haproxy needs a combined PEM.